lerobot/common/robot_devices/robots/manipulator.py

```diff
@@ -497,15 +515,21 @@ def teleop_step(
             # Cap goal position when too far away from present position.
             # Slower fps expected due to reading from the follower.
             if self.config.max_relative_target is not None:
-                present_pos = self.follower_arms[name].read("Present_Position")
+                if self.robot_type == "u850":
```

Comment:
This approach of adding if/elif throughout the codebase doesn't seem scalable.

What do folks think about creating an abstract base class that all robots can implement their wrapper API's within, so we don't have to edit so much code in a brittle way to add new robot support?

I can take this on- I'm currently working on adding support for the elephant robotics MyArmM&C series.

Reply:
Okay, sounds good! Maybe let's keep things as is, once xarm / myarm support is merged in I can jump in and write wrappers for all the currently supported robots.

The only hard thing will just be to have everyone test their bots out for regressions.